projects/plutus/inc/ltc_afe.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+// driver for LTC6804-1 AFE chip
+// requires GPIO, Interrupts and Soft Timers to be initialized
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include "gpio.h"
+#include "spi.h"
+#include "status.h"
+
+// static_assert shim macro since we aren't using C11
+// taken from: http://www.pixelbeat.org/programming/gcc/static_assert.html
+#define ASSERT_CONCAT_(a, b) a##b
+#define ASSERT_CONCAT(a, b) ASSERT_CONCAT_(a, b)
+/* These can't be used after statements in c89. */
+#ifdef __COUNTER__
+  #define STATIC_ASSERT(e, m) \
+    ; enum { ASSERT_CONCAT(static_assert_, __COUNTER__) = 1/(int)(!!(e)) }
+#else
+  /* This can't be used twice on the same line so ensure if using in headers
+   * that the headers are not included twice (by wrapping in #ifndef...#endif)
+   * Note it doesn't cause an issue when used on same line of separate modules
+   * compiled with gcc -combine -fwhole-program.  */
+  #define STATIC_ASSERT(e, m) \
+    ; enum { ASSERT_CONCAT(assert_line_, __LINE__) = 1/(int)(!!(e)) }
+#endif
+
+#define SWAP_UINT16(x) (uint16_t)(((uint16_t)(x) >> 8) | ((uint16_t)(x) << 8))
```

Comment:
Yes. In fact, the format of the registers uses both big and little endian (some of it is mentioned on p.44, and other bits are scattered throughout the datasheet). The **TL;DR** is basically:

* 12-bit, 16-bit and 24-bit values are little endian
* commands and PEC are big endian